crates/amalthea/src/socket/shell.rs

```diff
@@ -145,7 +147,7 @@ impl Shell {
                 self.handle_request(req, |h, r| self.handle_comm_info_request(h, r))
```

Comment:
Your change to remove `handle_request()` from `handle_comm_msg()` below implies we could also do it here with `handle_comm_info_request()`, because it doesn't use the `shell_handler` either?

But unlike `handle_comm_msg()`, `handle_comm_info_request()` _does_ have a return value it needs to send back. So that makes things complicated.